File: 0045-jump-game-ii/0045-jump-game-ii.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int jump =0,far =0,end=0;
        for(int i=0;i<nums.size()-1;i++){
            if(i>far)return 0 ;
            far = max(far,i+nums[i]);
            if(i==end){
                jump++;
                end=far;
            }
        }
        return jump;
    }
};

File: 0045-jump-game-ii/0045-jump-game-ii.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        maxi = 0
        jump = 0
        end = 0
        for i in range(len(nums)-1):
            maxi = max(maxi,i+nums[i])
            if  i ==end:
                jump +=1
                end =maxi
                if maxi>=(len(nums)-1):
                    break
        return jump